cognee/modules/users/get_user_manager.py

```
@@ -43,6 +45,31 @@ async def on_after_request_verify(
+            self.password_helper.hash(credentials.password)
+            return None
+
+        try:
+            verified, updated_password_hash = self.password_helper.verify_and_update(
+                credentials.password, user.hashed_password
+            )
+        except UnknownHashError:
+            raise HTTPException(
+                status_code=400,
+                detail="This user does not have a password. Use API key authentication.",
+            )
```

Ruff (0.15.14)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)